### PR TITLE
feat: query for payment pointer with channel URL

### PR DESCRIPTION
**Pull request by njlie**
Extension side of [coilhq/coil #2939](https://github.com/coilhq/coil/issues/2939).

### CODE REVIEW

**Comment by sublimator** [comment]
>  coil-puppeteer-utils:test test page failed to open stream. page= twitchUrl +1ms

Seems like the twitch test failed ?

File: packages/coil-extension/src/content/services/RunContentHandler.ts [hunk 1]
```diff
@@ -16,6 +16,12 @@ interface GetPageData {
   }
 }
 
+interface HyperlinkElement extends Element {
```
Comment (sublimator) [inline packages/coil-extension/src/content/services/RunContentHandler.ts]: There's HTMLAnchorElement

File: packages/coil-extension/src/content/services/RunContentHandler.ts [hunk 2]
```diff
@@ -28,18 +34,43 @@ export class RunContentHandler {
     private client: GraphQlClient
   ) {}
 
-  async adaptedPageDetails(url: string) {
+  async adaptedPageDetails(url: string, site: string) {
     debug('fetching payment pointer for this page', url)
 
+    // Wait for page to load to ensure there's a channel url to grab
+    if (document.readyState !== 'complete' && site === 'youtube') {
```
Comment (sublimator) [inline packages/coil-extension/src/content/services/RunContentHandler.ts]: Try and use document/window via DI so as this gets restructured it's in a good shape for tests with mocking

File: packages/coil-extension/src/content/services/RunContentHandler.ts [hunk 3]
```diff
@@ -16,6 +16,12 @@ interface GetPageData {
   }
 }
 
+interface HyperlinkElement extends Element {
+  href: string
+}
+
+const CHANNEL_ID_REGEX = /https?:\/\/(www\.)?youtube\.com\/channel\//i
```
Comment (sublimator) [inline packages/coil-extension/src/content/services/RunContentHandler.ts]: This appears unused ?

File: packages/coil-extension/src/content/services/RunContentHandler.ts [hunk 4]
```diff
@@ -24,22 +29,40 @@ export class RunContentHandler {
...
+    debug('fetching payment pointer for this page', url, site)
+
+    const variables = { url }
+
+    if (site === 'youtube') {
+      // Wait for page to load and the element containing the channelId is
+      // available
+
+      await this.windowLoaded()
+      const channelUrl = await this.pollForYouTubeChannelId({
```
Comment (sublimator) [inline packages/coil-extension/src/content/services/RunContentHandler.ts]: @njlie 
I pushed a commit while I was testing this out. Seems sometimes the element isn't available until a bit after load.

File: packages/coil-extension/src/content/services/RunContentHandler.ts [hunk 5]
```diff
@@ -54,6 +77,47 @@ export class RunContentHandler {
...
+
+  async windowLoaded() {
+    if (this.document.readyState !== 'complete') {
+      await new Promise(resolve => {
+        this.window.addEventListener(
+          'load',
+          () => {
+            resolve()
+          },
+          { once: true }
```
Comment (sublimator) [inline packages/coil-extension/src/content/services/RunContentHandler.ts]: @njlie 
Probably doesn't *really* matter but could potentially add a bunch of persistent listeners given youtube uses HTML5 pushState so can use once:true

File: packages/coil-extension/src/content/services/RunContentHandler.ts (hunk 3 above)
Comment (njlie) [reply packages/coil-extension/src/content/services/RunContentHandler.ts]: Yeah, was in an earlier attempt that I forgot to remove.

**Comment by njlie** [comment]
> > coil-puppeteer-utils:test test page failed to open stream. page= twitchUrl +1ms
> 
> Seems like the twitch test failed ?
> 
> edit: is that because it requires backend features that aren't deployed to production yet ? the new graphql query?

Yeah, we're gonna need to roll out some API changes first before this will work.

Also, thanks for updating the branch.

File: packages/coil-extension/src/content/services/DomService.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        await timeout(everyMs)
+      }
+    }
+    return null
+  }
+
+  async windowLoaded() {
+    if (this.document.readyState !== 'complete') {
+      await new Promise(resolve => {
+        this.window.addEventListener(
```
Comment (sublimator) [inline packages/coil-extension/src/content/services/DomService.ts]: wait, what about document listener for DOMContentLoaded ?

**Pull request merged**